api/src/test/java/gov/cms/ab2d/api/controller/BulkDataAccessAPIIntegrationTests.java

```diff
@@ -98,4 +106,85 @@ public void testPatientExportWithInvalidOutputFormat() throws Exception {
+        job.setStatus(JobStatus.SUCCESSFUL);
+        job.setProgress(100);
+        LocalDateTime expireDate = LocalDateTime.now().plusDays(100);
+        job.setExpires(expireDate);
+        LocalDateTime now = LocalDateTime.now();
+        job.setCompletedAt(now);
+
+        JobOutput jobOutput = new JobOutput();
+        jobOutput.setFhirResourceType("ExplanationOfBenefits");
+        jobOutput.setJob(job);
+        jobOutput.setFilePath("http://localhost/some/path/file.ndjson");
+        job.getJobOutput().add(jobOutput);
+
+        JobOutput errorJobOutput = new JobOutput();
+        errorJobOutput.setFhirResourceType(OPERATION_OUTCOME);
+        errorJobOutput.setJob(job);
+        errorJobOutput.setFilePath("http://localhost/some/path/error.ndjson");
+        errorJobOutput.setError(true);
+        job.getJobOutput().add(errorJobOutput);
+
+        jobRepository.saveAndFlush(job);
+
+        this.mockMvc.perform(get(statusUrl).contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().is(200))
+                .andExpect(header().string("Expires", DateUtil.formatLocalDateTimeAsUTC(expireDate)))
+                .andExpect(jsonPath("$.transactionTime", Is.is(new DateTimeType(DateUtil.convertLocalDateTimeToDate(now)).toHumanDisplay())))
+                .andExpect(jsonPath("$.request", Is.is(job.getRequestURL())))
+                .andExpect(jsonPath("$.requiresAccessToken", Is.is(true)))
+                .andExpect(jsonPath("$.output[0].type", Is.is("ExplanationOfBenefits")))
+                .andExpect(jsonPath("$.output[0].url", Is.is("http://localhost/some/path/file.ndjson")))
+                .andExpect(jsonPath("$.error[0].type", Is.is(OPERATION_OUTCOME)))
+                .andExpect(jsonPath("$.error[0].url", Is.is("http://localhost/some/path/error.ndjson")));
+    }
+
+    @Test
+    public void testGetStatusWhileFailed() throws Exception {
+        MvcResult mvcResult = this.mockMvc.perform(get(API_PREFIX + PATIENT_EXPORT_PATH + "?_type=ExplanationOfBenefits").contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+        String statusUrl = mvcResult.getResponse().getHeader("Content-Location");
+
+        Job job = jobRepository.findAll(Sort.by(Sort.Direction.DESC, "id")).iterator().next();
+        job.setStatus(JobStatus.FAILED);
+        LocalDateTime expireDate = LocalDateTime.now().plusDays(100);
+        job.setExpires(expireDate);
+
+        jobRepository.saveAndFlush(job);
+
+        this.mockMvc.perform(get(statusUrl).contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().is(500))
+                .andExpect(jsonPath("$.resourceType", Is.is("OperationOutcome")))
+                .andExpect(jsonPath("$.issue[0].severity", Is.is("error")))
+                .andExpect(jsonPath("$.issue[0].code", Is.is("invalid")))
+                .andExpect(jsonPath("$.issue[0].details.text", Is.is("JobProcessingException: Job failed while processing")));
+    }
```

Comment:
Need a test for 404 (job not found). Also need a test for an empty Job ID passed in the URl

Reply:
Added in those tests